pkg/chunk/chunk_store_utils.go

```diff
@@ -125,7 +125,7 @@ func NewChunkFetcher(cfg cache.Config, cacheStubs bool, storage ObjectClient) (*
 func (c *Fetcher) Stop() {
 	close(c.decodeRequests)
 	c.wait.Wait()
-	c.cache.Stop()
+	_ = c.cache.Stop()
```

Comment:
So if we want to handle this error I would think instead of logging it here we should let it flow up the stack.

Stop() gets called here which has a return of error
https://github.com/cortexproject/cortex/blob/ab3e8360fe30150a516448e4c3bae596dc4d8a17/pkg/cortex/modules.go#L284

Which is defined in the module definition for what "stop" should do.
https://github.com/cortexproject/cortex/blob/ab3e8360fe30150a516448e4c3bae596dc4d8a17/pkg/cortex/modules.go#L511-L527

And then as modules are stopped if one errors we log it at that level
https://github.com/cortexproject/cortex/blob/ab3e8360fe30150a516448e4c3bae596dc4d8a17/pkg/cortex/cortex.go#L265-L272

But currently the hinderance on easily doing that is the Worker interface does not allow error to be returned.
https://github.com/cortexproject/cortex/blob/ab3e8360fe30150a516448e4c3bae596dc4d8a17/pkg/querier/frontend/worker.go#L49-L51

So do you have any feelings on updating that interface to allow for that?? My gut says maybe we split that out into its own issue and leave this as is since right now it's no worse than it currently is on master.

Reply:
I vote to remove the error return - #2030

Reply:
@zendern I've merged the PR https://github.com/cortexproject/cortex/pull/2030. May you rebase `master`, please?